underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSInputStream.java

```diff
@@ -147,21 +134,55 @@ public int read(byte[] b, int off, int len) throws IOException {
    */
   @Override
   public long skip(long n) throws IOException {
+    if (n <= 0) {
+      return 0;
+    }
     if (mInputStream.available() >= n) {
       return mInputStream.skip(n);
     }
-    // The number of bytes to skip is possibly large, open a new stream from GCS.
-    mInputStream.close();
+    closeStream();
     mPos += n;
-    try {
-      mObject = mClient.getObject(mBucketName, mKey, null /* ignore ModifiedSince */,
-          null /* ignore UnmodifiedSince */, null /* ignore MatchTags */,
-          null /* ignore NoneMatchTags */, mPos /* byteRangeStart */,
-          null /* ignore byteRangeEnd */);
-      mInputStream = new BufferedInputStream(mObject.getDataInputStream());
-    } catch (ServiceException e) {
-      throw new IOException(e);
-    }
+    openStream();
     return n;
   }
+
+  /**
+   * Opens a new stream at mPos if the wrapped stream mInputStream is null.
+   */
+  private void openStream() throws IOException {
+    ServiceException lastException = null;
+    while (mRetryPolicy.attempt()) {
+      try {
+        GSObject object;
+        if (mPos > 0) {
+          object = mClient.getObject(mBucketName, mKey, null, null, null, null, mPos, null);
+        } else {
+          object = mClient.getObject(mBucketName, mKey);
+        }
+        mInputStream = new BufferedInputStream(object.getDataInputStream());
+        return;
+      } catch (ServiceException e) {
+        LOG.warn("Attempt {} to open key {} in bucket {} failed with exception : {}",
```

Comment:
should we include `mPos` in the message?

Reply:
Added the mPos information